src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/repositories/FermaBfdSessionRepository.java

```diff
@@ -0,0 +1,92 @@
+/* Copyright 2020 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.ferma.repositories;
+
+import org.openkilda.model.BfdSession;
+import org.openkilda.model.BfdSession.BfdSessionData;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.ConstraintViolationException;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.ferma.FramedGraphFactory;
+import org.openkilda.persistence.ferma.frames.BfdSessionFrame;
+import org.openkilda.persistence.ferma.frames.KildaBaseVertexFrame;
+import org.openkilda.persistence.ferma.frames.converters.SwitchIdConverter;
+import org.openkilda.persistence.repositories.BfdSessionRepository;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Ferma (Tinkerpop) implementation of {@link BfdSessionRepository}.
+ */
+class FermaBfdSessionRepository extends FermaGenericRepository<BfdSession, BfdSessionData, BfdSessionFrame>
+        implements BfdSessionRepository {
+    FermaBfdSessionRepository(FramedGraphFactory<?> graphFactory, TransactionManager transactionManager) {
+        super(graphFactory, transactionManager);
+    }
+
+    @Override
+    public Collection<BfdSession> findAll() {
+        return framedGraph().traverse(g -> g.V()
+                .hasLabel(BfdSessionFrame.FRAME_LABEL))
+                .toListExplicit(BfdSessionFrame.class).stream()
+                .map(BfdSession::new)
+                .collect(Collectors.toList());
```

Comment:
Why no to return Stream<?> in such methods? (just because we already have it and most times caller needs to make only one iteration over objects in response... so converting it into List is resource wasting)

Reply:
It's a good point. When the initial implementation of the persistence layer was introduced, we inherited the Neo4j-OGM approach - fetching via collections. 
However, I don't think this task is a right time to refactor the api.